pysubtools/subtitle.py

```diff
@@ -8,625 +8,677 @@
+    def to_yaml(cls, dumper, data):
+        if isinstance(data, (int, float)):
+            data = cls.from_seconds(data)
+
+        try:
+            return dumper.represent_scalar('!human_time', unicode(data))
+        except NameError:
+            return dumper.represent_scalar('!human_time', str(data))
+
+    @classmethod
+    def from_seconds(cls, time):
+        obj = cls()
+        time = float(time)
+        obj.hours = int(time // 3600)
+        time -= obj.hours * 3600
+        obj.minutes = int(time // 60)
+        time -= obj.minutes * 60
+        obj.seconds = time
+        return obj
+
+    @classmethod
+    def from_string(cls, time):
+        obj = cls()
+
+        try:
+            is_str = isinstance(time, basestring)
+        except NameError:
+            is_str = isinstance(time, str)
+
+        if is_str:
+            time = time.split(':')
+            obj.hours = int(time[0])
+            obj.minutes = int(time[1])
+            obj.seconds = float(time[2])
+        else:
+            raise TypeError("Unknown time format.")
 
-class SubtitleLine(UnicodeMixin, object):
-  """
-  Class representing a line inside SubtitleUnit. It acts as an ordinary
-  unicode objects, but has an ability to store additional metadata.
-  """
-  # Unhashable
-  __hash__ = None
-
-  def __init__(self, text, **kwargs):
-    self.text = text
-    # Update with additional metadata
-    self.__dict__.update(kwargs)
-
-  def export(self):
-    """Returns line in format for export."""
-    output = dict(self.__dict__)
-    text = output.pop('text', '')
-    if not output:
-      output = text
-    else:
-      output['text'] = text
-    return output
-
-  @classmethod
-  def from_export(cls, obj):
-    return cls(**obj)
-
-  def __unicode__(self):
-    return self.text
-
-  if sys.version_info[0] >= 3: # Python 3
-    def __repr__(self):
-      return "SubtitleLine({}{})".format(
-        self.text,
-        (', ' + ', '.join([' = '.join([k, str(v)]) for k, v in self.meta.items()])) if self.meta else ''
-      )
-  else:  # Python 2
-    def __repr__(self):
-      return "SubtitleLine({}{})".format(
-        self.text,
-        (', ' + ', '.join([' = '.join([k, unicode(v)]) for k, v in self.meta.items()])) if self.meta else ''
-      ).encode('utf8')
-
-  def __eq__(self, other):
-    if not isinstance(other, SubtitleLine):
-      return False
-    return self.__dict__ == other.__dict__
-
-  def __len__(self):
-    return len(self.text)
-
-  @property
-  def meta(self):
-    d = dict(self.__dict__)
-    # Remove important part of metadata
-    d.pop('text')
-    return d
+        return obj
 
-class SubtitleLines(list):
-  """Modified list class for special tratment of lines."""
-  __slots__ = ()
+    def __unicode__(self):
+        return '{:02d}:{:02d}:{:06.3f}'.format(self.hours,
+                                               self.minutes,
+                                               self.seconds)
 
-  def __new__(cls, l = []):
```

Comment:
I think `l` is meant as `lines`

Reply:
Yes, but it threw a warning or error, I can't remember which one, I can still change it back to 'l' of course

Reply:
Probably shadowing variable.

In this case, I saw a little trick to append the variable with `_`. For instance `lines_`.

Reply:
Agreed, this is the most clear solution